### PR TITLE
Add Share Component

### PR DESCRIPTION
**Pull request by estacioneto**
#### What is the purpose of this pull request?

Add Share Component.

#### What problem is this solving?

Need to share the page.

#### How should this be manually tested?

[Access the workspace](https://share--storecomponents.myvtex.com/google-pixel2/p)

#### Screenshots or example usage

image

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Share/utils/socialResolver.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import * as ReactShare from 'react-share'
+
+import { SOCIAL_ENUM_TO_COMPONENT } from '../constants/social'
+
+export function SocialButton({ url, size, socialEnum }) {
```
Comment (lucasecdb) [inline react/components/Share/utils/socialResolver.js]: why not `export default`?

File: react/components/Share/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+```html
+<Share social={{'Facebook': true, 'WhatsApp': true }} />
+```
+
+| Prop name      | Type      | Description                                                                |
+| -------------- | --------- | -------------------------------------------------------------------------- |
+| `social`       | `Object!` | Social networks definition. The key is the name and the value is a boolean |
+| `options`      | `Object`  | Object to configure the Component                                          |
+| `options.size` | `Number`  | Buttons size                                                               |
```
Comment (andreldsa) [inline react/components/Share/README.md]: pixels? What's the metric?

File: react/components/Share/utils/socialResolver.js (hunk 1 above)
Comment (andreldsa) [reply react/components/Share/utils/socialResolver.js]: why?

Comment (estacioneto) [reply react/components/Share/utils/socialResolver.js]: I used this so I can export the component but also, in the future, export social resolvers functions. That's why the file name is this one.

Comment (estacioneto) [reply react/components/Share/utils/socialResolver.js]: And I created this component to help to selecting the social button. To divide this logic.

**Comment by brunojdo** [comment]
It will be awesome if we can create a template message (via schema) something like: "Compre {productName} por {productPrice} https://share--storecomponents.myvtex.com/google-pixel2/p" na {storeName}

**Pull request merged**